modules/internal/ChapelArray.chpl

```diff
@@ -799,15 +814,37 @@ module ChapelArray {
   }
 
   proc chpl__eltTypeFromArrayRuntimeType(type rtt) type {
-    pragma "unsafe"
-    var arr: rtt;
-    return arr.eltType;
+    pragma "ignore runtime type"
+    proc getArrEltType() type {
+      pragma "unsafe"
+      var arr : rtt;
+      return __primitive("static typeof", arr.eltType);
+    }
+
+    // does the element type have a runtime component?
```

Comment:
Do you need to add a case when the element type is a domain?

Reply:
It already is there, on line 826